BlueDBOnDisk/src/main/java/org/bluedb/disk/backup/BackupManager.java

```diff
@@ -58,39 +82,124 @@ private void copyMetaData(ReadWriteCollectionOnDisk<?> collection, Path tempFold
+		Path dst = translatePath(dbPath, tempFolder, src);
+		FileUtils.ensureDirectoryExists(dst.toFile());
+		boolean isFileEmpty = true;
+		
+		try(BlueObjectOutput<BlueEntity<?>> output = BlueObjectOutput.createWithoutLockOrSerializer(dst)) {
+			try(BlueObjectInput<?> fileInputStream = segment.getObjectInputFor(groupingNumber)) {
+				while(fileInputStream.hasNext()) {
+					BlueEntity<?> next = (BlueEntity<?>) fileInputStream.next();
+					if(next.getKey().isInRange(includedTimeRange.getStart(), includedTimeRange.getEnd())) {
+						output.writeBytes(fileInputStream.getLastBytes());
+						isFileEmpty = false;
+					}
+				}
+			}
+		}
+		
+		if(isFileEmpty) {
+			FileUtils.deleteIfExistsWithoutLock(dst);
+		}
+	}
+
+	private void copyDataFileStraightOver(ReadWriteSegment<?> segment, Path tempFolder, long groupingNumber) throws BlueDbException {
+		try (BlueReadLock<Path> lock = segment.getReadLockFor(groupingNumber)) {
+			Path src = lock.getKey();
+			Path dst = translatePath(dbPath, tempFolder, src);
+			FileUtils.copyFileWithoutLock(src, dst);  // already have read lock on src, shouldn't need write lock on dst
+		}
+	}
+
+	private void copyChanges(ReadWriteCollectionOnDisk<?> collection, Range includedTimeRange, Range includedChangeTimes, Path tempFolder) throws BlueDbException {
 		RecoveryManager<?> recoveryManager = collection.getRecoveryManager();
-		List<File> changesToCopy = recoveryManager.getChangeHistory(backupStartTime, backupEndTime);
+		List<File> changesToCopy = recoveryManager.getChangeHistory(includedChangeTimes.getStart(), includedChangeTimes.getEnd());
 		Path historyFolderPath = recoveryManager.getHistoryFolder();
 		Path destinationFolderPath = translatePath(dbPath, tempFolder, historyFolderPath);
 		destinationFolderPath.toFile().mkdirs();
 		for (File file: changesToCopy) {
 			Path destinationPath = Paths.get(destinationFolderPath.toString(), file.getName());
-			FileUtils.copyFileWithoutLock(file.toPath(), destinationPath);
+			if(!includedTimeRange.isMaxRange() && collection.isTimeBased()) {
+				copyChangeAfterFilteringBasedOnTime(collection, includedTimeRange, recoveryManager, file, destinationPath);
+			} else {
+				copyChangeStraightOver(recoveryManager, file, destinationPath);
+			}
+		}
+	}
+
+	private void copyChangeAfterFilteringBasedOnTime(ReadWriteCollectionOnDisk<?> collection, Range includedTimeRange, RecoveryManager<?> recoveryManager, File file, Path destinationPath) throws BlueDbException {
+		Recoverable<?> change = (Recoverable<?>) collection.getFileManager().loadObject(file);
+		if(shouldCopyChange(includedTimeRange, change)) {
+			collection.getFileManager().saveObject(destinationPath, change);
 			recoveryManager.markChangePending(destinationPath);
 		}
 	}
 
+	private boolean shouldCopyChange(Range includedTimeRange, Recoverable<?> change) {
+		if(change instanceof PendingChange) {
+			if(((PendingChange<?>)change).getKey().isInRange(includedTimeRange.getStart(), includedTimeRange.getEnd())) {

```

Comment:
Might consider making a helper method BlueKey.isInRange(TimeRange) because you do this at least twice.

Reply:
Hey Jeremy what's up? Always glad to see your comments! I just tried to implement this comment but BlueKey is an interface in the API project and it doesn't know about the Range class so I don't think I'm going to worry too much about it for now.

Reply:
:) How goes it?   Yeah, I was waiting for a build so I figured I'd scroll through and take a brief look.